### PR TITLE
Actually fix redisson instrumentation after 3.16.8 release

### PR DESCRIPTION
**Pull request by mateuszrzeszutek**
Implemented Lauri's idea of using `VirtualField`s to avoid calling `getPromise()`

### CODE REVIEW

**Review by breedx-splk — APPROVED** [review]
Looks good to me. It took me a bit to figure out the `CommandData` vs. `CommandsData` difference. Does it make sense to combine those into a single class that takes the class and name? Might be a little clearer. 🤷🏻

**Comment by mateuszrzeszutek** [comment]
You still need to have different advice classes for these two - except the method matcher there's pretty much nothing in common.

**Review by laurit — APPROVED** [review]

**Comment by breedx-splk** [comment]
> You still need to have different advice classes for these two - except the method matcher there's pretty much nothing in common.

Ah yeah, didn't consider that we still need two advices because we don't control creation of them (and just pass the name as a string). Gotcha.

File: instrumentation/redisson-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/RedissonInstrumentationModule.java [hunk 1]
```diff
@@ -21,6 +21,9 @@ public RedissonInstrumentationModule() {
 
   @Override
   public List<TypeInstrumentation> typeInstrumentations() {
-    return singletonList(new RedisConnectionInstrumentation());
+    return asList(
+        new CommandDataInstrumentation(),
+        new CommandDataInstrumentation(),
```
Comment (trask) [inline instrumentation/redisson-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/RedissonInstrumentationModule.java:26]:
```suggestion
        new CommandsDataInstrumentation(),
```

**Review by trask — COMMENTED** [review]
what do you think about using reflection to call `getPromise()` instead?

**Comment by mateuszrzeszutek** [comment]
> what do you think about using reflection to call `getPromise()` instead?

That could work too! I reworked this PR to use `MethodHandle`s instead

File: instrumentation/redisson-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/RedissonRequest.java [hunk 2]
```diff
@@ -23,6 +29,11 @@
 @AutoValue
 public abstract class RedissonRequest {
 
+  private static final VirtualField<CommandData<?, ?>, CompletionStage<?>> commandDataPromiseField =
```
Comment (anuraaga) [inline instrumentation/redisson-3.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/redisson/RedissonRequest.java:32]: Think can remove these

**Review by anuraaga — APPROVED** [review]

**Review by trask — APPROVED** [review]

**Pull request merged**